proofs/eo/cpc/rules/Quantifiers.eo

```diff
@@ -80,3 +80,58 @@
   :requires ((($beta_reduce a @list.nil) b))
   :conclusion (= a b)
 )
+
+;;;;; ProofRule::QUANT_VAR_REORDERING
+
+; rule: quant_var_reordering
+; implements: ProofRule::QUANT_VAR_REORDERING
+; premises:
+; - C1: The clause to reorder.
+; args:
+; - C2 Bool: The desired reordered form of C1.
+; requires: Showing that C1 can be reordered to obtain C2.
+; conclusion: The desired clause C2.
+(declare-rule quant_var_reordering ((x @List) (y @List) (F Bool))
+    :args ((= (forall x F) (forall y F)))
+    :requires ((($nary_is_eq_set @list @list.nil x y) true))
```

Comment:
Actually this is a bit of an interesting question:
(1) Does the Eunoia definition serve to document *exactly* what the CPC rule does? or
(2) Does the Eunoia definition serve to document a *possible* (perhaps more permissive) way to implement the CPC rule? 

In this case it is sound to allow repetitions.

Note that a similar question happened for REORDERING, where I decided on (2) https://github.com/cvc5/cvc5/blob/main/proofs/eo/cpc/rules/Booleans.eo#L194 since it is sound to only check subset, despite the internal proof checker being more pedantic.